Sources/ArcGISToolkit/FloorFilter/FloorFilterViewController.swift

```diff
@@ -0,0 +1,536 @@
+                _selectedSite = selectedLevelsFacility?.site
+            }
+            viewModel.selectedLevel = newValue
+            viewModel.filterMapToSelectedLevel()
+        }
+    }
+    
+    /// Listener when a level is changed
+    public var onSelectedLevelChangedListener : (() -> Void)? = nil
+
+    /// Refresh the view with the new map
+    public func refresh(geoView: AGSGeoView?){
+        self.geoView = geoView
+        state = FloorFilterState.initiallyCollapsed
+        updateViewsVisibilityForState(state: state)
+    }
+    
+    /// Variables for styling the Floor Filter View
+    public var fontSize: CGFloat = 14.0
+    public var fontName: String = "Avenir"
+    public var selectionColor: UIColor = UIColor(hexString: "#C7EAFF")
+    public var backgroundColor: UIColor = UIColor(hexString: "#F2F2F7")
+    public var selectedTextColor: UIColor = UIColor(hexString: "#004874")
+    public var unselectedTextColor: UIColor = UIColor(hexString: "#323236")
+    
+    /// Floor Filter UI Elements and Constraints
+    @IBOutlet var floorFilterView: UIView!
+    @IBOutlet weak var siteBtn: UIButton!
+    @IBOutlet weak var levelsTableView: UITableView!
+    @IBOutlet weak var closeBtn: UIButton!
+    @IBOutlet weak var floorFilterStackView: UIStackView!
+    @IBOutlet weak var closeBtnHeight: NSLayoutConstraint!
+    @IBOutlet weak var siteBtnHeight: NSLayoutConstraint!
+    @IBOutlet weak var tableViewHeight: NSLayoutConstraint!
+    @IBOutlet weak var closeBtnWidth: NSLayoutConstraint!
+    @IBOutlet weak var siteBtnWidth: NSLayoutConstraint!
+    @IBOutlet weak var levelCellWidth: NSLayoutConstraint!
+    
+    /// Floor Filter view styles variables
+    /// These are the default, but will get updated with values that are provided during initialization
+    private var buttonHeight: CGFloat = 35
+    private var buttonWidth: CGFloat = 40
+    private var maxDisplayLevels: Int = 3
+  
+    private var delegate: FloorFilterViewControllerDelegate?
+    private var viewModel = FloorFilterViewModel()
+    
+    /// State of the visibility of the Floor Filter
+    private enum FloorFilterState {
+        case initiallyCollapsed
+        case partiallyExpanded
+        case fullyExpanded
+    }
+    private var state = FloorFilterState.initiallyCollapsed
+    
+    private var floorManager: AGSFloorManager?
+    
+    /// GeoView that the Floor Filter is rendered on
+    /// For Version 1 only MapView (2D) is supported to render the Floor Filter
+    private var geoView: AGSGeoView? {
+        didSet {
+            if geoView != nil {
+                if let mapView = geoView as? AGSMapView {
+                    mapView.map?.load { [weak self] (_) in
+                        self?.viewModel.mapView = mapView
+                        self?.viewModel.map = mapView.map
+                        self?.floorManager = mapView.map?.floorManager
+                        self?.initializeFloorManager()
```

Comment:
I'd use a `guard` statement here so you don't have to continually unwrap `self`.
```suggestion
                        guard let self = self else { return }
                        self.viewModel.mapView = mapView
                        self.viewModel.map = mapView.map
                        self.floorManager = mapView.map?.floorManager
                        self.initializeFloorManager()
```